File: linux/ardium_source/presentation/02_Recursion/recursion.py
# Topic: Recursion
# Calculate Fibonacci(10)

def fib(n):
    if n <= 1:
        return n
    return fib(n - 1) + fib(n - 2)

def main():
    result = fib(10)
    print(f"Fibonacci(10) = {result}")

if __name__ == "__main__":
    main()

File: presentation/02_Recursion/recursion.m
// Topic: Recursion
// Calculate Fibonacci(10)

#import <Foundation/Foundation.h>

int fib(int n) {
    if (n <= 1) {
        return n;
    }
    return fib(n - 1) + fib(n - 2);
}

int main(int argc, const char * argv[]) {
    @autoreleasepool {
        int result = fib(10);
        NSLog(@"Fibonacci(10) = %d", result);
    }
    return 0;
}
